### PR TITLE
[CAI-95] fastapi docker

### PR DESCRIPTION
**Pull request by batdevis**
#### List of Changes
Fastapi project with docker ready for lambda function

#### Motivation and Context
`Frontend -> API in lambda -> Llamaindex -> bedrock`

#### Checklist:

- [x] we need a container registry
- [ ] add the docker build and push commands in the pipeline

### CODE REVIEW

File: apps/infrastructure/src/modules/chatbot/variables.tf [hunk 1]
```diff
@@ -34,4 +34,35 @@ variable "aws_chatbot_region" {
...
   type        = string
   description = "The name of the website bucket"
-}
\ No newline at end of file
+}
+
+variable "chatbot_lambda_image_container_url" {
+  type        = string
+  description = "Chatbot Lambda image container URL"
+}
```
Comment (christian-calabrese) [inline apps/infrastructure/src/modules/chatbot/variables.tf:42]: Can be removed when the ECR is created via terraform

File: apps/infrastructure/src/modules/chatbot/lambda_chatbot.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  function_name = "${random_pet.this.id}-chatbot-api-tf"
+  description   = "chatbot"
+
+  environment_variables = local.lambda_env_variables
+  create_package = false
+
+  package_type  = "Image"
+  architectures = ["x86_64"]
+
+  image_uri = var.chatbot_lambda_image_container_url
```
Comment (christian-calabrese) [inline apps/infrastructure/src/modules/chatbot/lambda_chatbot.tf:24]: Can reference the ECR when created via terraform

File: apps/chatbot/src/modules/create_vector_index.py [hunk 3]
```diff
@@ -38,7 +40,8 @@
         embed_model,
         documentation_dir=params["documentation"]["path"],
         save_dir=params["vector_index"]["path"],
-        s3_bucket_name=os.getenv("AWS_S3_BUCKET"),
+        s3_bucket_name=AWS_S3_BUCKET,
+        region=AWS_DEFAULT_REGION,
```
Comment (mdciri) [inline apps/chatbot/src/modules/create_vector_index.py:44]: the input `region` in the function `build_automerging_index()` was removed

File: .gitignore [hunk 4]
```diff
@@ -403,6 +403,7 @@ apps/chatbot/src/*/__pycache__
 apps/chatbot/*.ipynb
 apps/chatbot/results
 apps/chatbot/test.py
+!apps/chatbot/docker
```
Comment (mdciri) [inline .gitignore:406]: why the '!'?

File: apps/chatbot/src/app/main.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  answer = chatbot.generate(query.question)
+
+  # TODO: dynamoDB integration
+  body = {
+    "id": "",
+    "sessionId": "",
+    "question": query.question,
+    "answer": answer,
+    "createdAt": query.queriedAt,
+    "queriedAt": query.queriedAt
```
Comment (mdciri) [inline apps/chatbot/src/app/main.py:54]: I wonder if we can add more here as: timestamp, latency, and context

File: .gitignore (hunk 4 above)
Comment (batdevis) [reply .gitignore:406]: in order to override
```
 390 # Docker config files                                                                                                                                                   
 391 docker/
```

**Review by mdciri — APPROVED** [review]

**Review by christian-calabrese — APPROVED** [review]
LGTM

**Pull request merged**